src/code_f4c60_len_4300.c

```diff
@@ -20,17 +28,56 @@ INCLUDE_ASM(code_f4c60_len_4300, SwitchMessage);
 
 INCLUDE_ASM(code_f4c60_len_4300, ShowChoice);
 
-INCLUDE_ASM(code_f4c60_len_4300, CloseChoice);
-
-INCLUDE_ASM(code_f4c60_len_4300, CancelMessage);
-
-INCLUDE_ASM(code_f4c60_len_4300, CancelMessageAndBlock);
-
-INCLUDE_ASM(code_f4c60_len_4300, SetMessageImages);
-
-INCLUDE_ASM(code_f4c60_len_4300, SetMessageString);
+s32 CloseChoice(script_context* script, s32 initialCall) {
+    close_message(D_802DB268);
+    return 1;
+}
+
+s32 CancelMessage(script_context* script, s32 initialCall) {
+    cancel_message(gCurrentPrintContext);
+    return 2;
+}
+
+s32 CancelMessageAndBlock(script_context* script, s32 initialCall) {
+    cancel_message(gCurrentPrintContext);
+    return 0;
+}
+
+s32 SetMessageImages(script_context* script, s32 initialCall) {
+    set_message_images(*script->ptrReadPos);
+    return 2;
+}
+
+s32 func_802D0C94(script_context* script, s32 initialCall) {
+    if (get_variable(script, *script->ptrReadPos) == 0) {
+        D_8009A650[0] |= 0x10;
+    } else {
+        D_8009A650[0] &= ~0x10;
+    }
+    return 2;
+}
+
+s32 SetMessageString(script_context* script, s32 initialCall) {
+    bytecode* ptrReadPos = script->ptrReadPos;
+    bytecode string = get_variable(script, *ptrReadPos++);
+    bytecode index = get_variable(script, *ptrReadPos++);
+
+    set_message_string(string, index);
+    return 2;
+}
 
 INCLUDE_ASM(code_f4c60_len_4300, SetMessageValue);
+// TODO: Figure out why there's an extra NOP after this function
```

Comment:
as discussed, this is most likely a C file split... please update the comment with a TODO if you would.